src/macaron/__main__.py

```diff
@@ -92,15 +93,18 @@ def analyze_slsa_levels_single(analyzer_single_args: argparse.Namespace) -> None
         branch = analyzer_single_args.branch
         digest = analyzer_single_args.digest
 
-        if repo_path and purl and not digest:
-            # To provide the purl together with the repository path, the user must specify the branch and commit
-            # digest.
-            logger.error(
-                "Please provide the commit digest for the repo at %s that matches to the PURL string %s.",
-                repo_path,
-                purl,
-            )
-            sys.exit(os.EX_USAGE)
+        if repo_path and purl:
+            # To provide the purl together with the repository path, the user must specify the commit digest unless the
+            # purl has a version.
+            purl_object = PackageURL.from_string(purl)
+            if not purl_object or not (purl_object.version or digest):
```

Comment:
Having looked at the [implementation of `PackageURL.from_string`](https://github.com/package-url/packageurl-python/blob/f7f41b89a941278e8f76c0aad3a9409c6583eda8/src/packageurl/__init__.py#L481), I don't think the returned object can be falsy or `None`. However, the function may raise `ValueError` which should be handled.